components/common/src/command/package/download.rs

```diff
@@ -0,0 +1,400 @@
+/// (regardless of `channel`). Providing a partially-qualified identifier will result in the
+/// installation of latest appropriate release from the given `channel`.
+///
+/// Any dependencies of will be retrieved from Builder (if they're not already cached locally).
+///
+/// At the end of this function, the specified package and all its dependencies will be downloaded
+/// on the system.
+
+/// Note: it's worth investigating whether
+/// LocalPackageUsage makes sense here
+/// Also, in the future we may want to accept an alternate builder to 'filter' what we pull down by
+/// That would greatly optimize the 'sync' to on prem builder case, as we could point to that
+/// and only fetch what we don't already have.
+#[allow(clippy::too_many_arguments)]
+pub fn start<U>(ui: &mut U,
+                url: &str,
+                channel: &ChannelIdent,
+                product: &str,
+                version: &str,
+                idents: Vec<PackageIdent>,
+                target: PackageTarget,
+                download_path: Option<&PathBuf>,
+                token: Option<&str>,
+                verify: bool)
+                -> Result<()>
+    where U: UIWriter
+{
+    debug!("Starting download with url: {}, channel: {}, product: {}, version: {}, target: {}, \
+            download_path: {:?}, token: {:?}, verify: {}",
+           url, channel, product, version, target, download_path, token, verify);
+
+    let key_download_path = &path_helper(download_path, "keys", &cache_key_path::<PathBuf>(None));
+    debug!("install key_download_path: {:?}", key_download_path);
+
+    let artifact_download_path = &path_helper(download_path,
+                                              "artifacts",
+                                              &cache_artifact_path::<PathBuf>(None));
+    debug!("install artifact_download_path: {:?}",
+           artifact_download_path);
+
+    // We deliberately use None to specifiy the default path as this is used for cert paths, which
+    // we don't want to override.
+    let api_client = Client::new(url, product, version, None)?;
+    let task = DownloadTask { idents,
+                              target,
+                              url,
+                              api_client,
+                              token,
+                              channel,
+                              artifact_download_path,
+                              key_download_path,
+                              verify };
+
+    let downloaded_artifacts: Vec<PackageArchive> = task.execute(ui).unwrap();
+
+    debug!("Expanded package count: {}", downloaded_artifacts.len());
+
+    Ok(())
+}
+
+struct DownloadTask<'a> {
+    idents: Vec<PackageIdent>,
+    target: PackageTarget,
+    url: &'a str,
+    api_client: BoxedClient,
+    token: Option<&'a str>,
+    channel: &'a ChannelIdent,
+    /// The path to the local artifact cache (e.g., /hab/cache/artifacts)
```

Comment:
Update comment... path to the artifact download location 
Alternatively remove comment as the name is self explanatory